html/reaction.html

```diff
@@ -807,114 +812,143 @@
-                  <tr><td>texture</td><td><div class="outcome_product_texture_type selector" data-proto="ReactionProduct_Texture_TextureType"></div></td></tr>
-                  <tr><td>details</td><td><div class="outcome_product_texture_details edittext"></div></td></tr>
-                </table>
-              </div>
-            </fieldset>
-
             <fieldset class="s5">
               <legend>
                 <span class="collapse"></span>
-                <span class="h5">Identity analyses</span>
-                <span data-toggle="tooltip" data-placement="right" title="Which analyses were used to confirm this product's identity.">
+                <span class="h5">Identifiers</span>
+                <span data-toggle="tooltip" data-placement="right" title="Compound identifiers should be structural (e.g., SMILES) whenever possible.">
                   <span class="fa fa-question-circle" aria-hidden="true"></span>
                 </span>
               </legend>
-              <div class="outcome_product_analysis_identities">
-              </div>
-              <div onclick="ord.products.addIdentity($(this).closest('.outcome_product'));" class="add">+ add identity analysis</div>
+              <!-- Compound rendering -->
+              <div class="component_rendering" style="text-align: center;"></div>
+              <div class="identifiers"></div>
+              <!-- Shortcuts for name resolving & drawing -->
+              <div onclick="ord.compounds.drawIdentifier($(this).closest('.component'));" class="add"><span class="far fa-edit" aria-hidden="true"></span> draw compound</div>
+              <div onclick="ord.compounds.addNameIdentifier($(this).closest('.component'));" class="add"><span class="fa fa-search" aria-hidden="true"></span> look up name</div>
+              <!-- End shortcuts -->
+              <div onclick="ord.compounds.addIdentifier($(this).closest('.component'));" class="add">+ add identifier</div>
             </fieldset>
 
+            <div id="product_measurement_template" class="product_measurement" style="display: none;">
+              <fieldset class="s6">
+                <legend>
+                  <span class="collapse"></span>
+                  <span class="h6">Product Measurement</span>
+                  <div onclick="ord.reaction.removeSlowly(this, '.product_measurement');" class="remove">remove</div>
+                  <span class="validate" onclick="ord.products.validateMeasurement($(this).closest('.product_measurement'))"></span>
+                </legend>
+                <table class="message">
+                  <tr>
+                    <td>based on</td>
+                    <td>
+                      <div class="product_measurement_analysis_key">
+                        <select class="analysis_key_selector">
+                          <option disabled selected value="">-- select an analysis label --</option>
```

Comment:
If you don't change from this value, the proto gets saved with an analysis key of "select an analysis label". This might be okay, but it's a little silly. There is an outcome-level validation error of an undefined analysis key.

Reply:
Fixed; now that selection does not populate the pbtxt.